zerver/tests/test_user_topics.py

```diff
@@ -283,3 +283,149 @@ def set_topic_visibility_for_user(user: UserProfile, visibility_policy: int) ->
             user_profile=cordelia, visibility_policy=UserTopic.VisibilityPolicy.UNMUTED
         )[0].last_updated
         self.assertEqual(cordelia_date_unmuted, date_unmuted)
+
+    @mock.patch("zerver.actions.user_topics.send_event")
```

Comment:
Why do we mock this here? The `with tornado_redirected_to_list` function is our standard way to skip Tornado logic and capture events... (I guess I see we use this pattern in `test_message_edit`, but I think we should plan to replace that with our standard API). 

I also wonder if `tornado_redirected_to_list` should be renamed to something like `with capture_send_event_calls()` or something -- that might be a good #backend topic for you to start. (I just had the thought that the name may not be super self-explanatory)

Reply:
> Why do we mock this here?

To verify if events are sent properly. Currently in `MutedTopicsTests`, we verify the event's content in [this line](https://github.com/zulip/zulip/blob/main/zerver/tests/test_user_topics.py#L109). We should instead verify if it works end-to-end, right?
(Let me know, if this approach looks good, then i will update the existing code)

> 'with tornado_redirected_to_list' function is our standard way to skip Tornado logic and capture events

used this. (I will update `test_message_edit` to follow this pattern in a separate PR)

> I also wonder if tornado_redirected_to_list should be renamed to something like with capture_send_event_calls() or something -- that might be a good #backend topic for you to start

Sounds better. (I will open a topic)

Reply:
Yeah, that's correct, sounds good!